### ISSUE TITLE
func swiftClassTypeFromString(className: String) -> AnyClass!  return nil

### ISSUE THREAD

**Issue by mengjiu-cao** [evermeer/EVReflection#19]
```
public class func swiftClassTypeFromString(className: String) -> AnyClass! {
```

//        if className.hasPrefix("Optional<") {
//            className = className.substringWithRange(Range<String.Index>(start: className.startIndex.advancedBy(9), end: className.endIndex.advancedBy(-1)))
//        }
        if className.hasPrefix("_TtC") {
            return NSClassFromString(className)
        }
        var classStringName = className
        if className.rangeOfString(".", options: NSStringCompareOptions.CaseInsensitiveSearch) == nil {
            let appName = getCleanAppName()
            classStringName = "(appName).(className)"
        }
        return NSClassFromString(classStringName)
    }

(lldb) po NSClassFromString("麦田.ImageTextContent")
nil

var appName = bundle.infoDictionary?["CFBundleName"] as? String ?? ""
appName is "麦田"

**Comment by mengjiu-cao** [comment]
when i drag the EVReflection module with WorkaroundSwiftGenericsTests in an empty project named EVReflection, swiftClassTypeFromString also return nil 

Printing description of classStringName:
"EVReflection.InstanceObject"
(lldb) po NSClassFromString(classStringName)
nil

(lldb)

**Comment by mengjiu-cao** [comment]
Printing description of classStringName:
"MaiTian.StatusModel<ImageTextContent>"
(lldb) po NSClassFromString(classStringName)
nil

(lldb) 

class StatusModel<T: EVObject>: StatusBaseModel, EVGenericsKVC {
    var content: T = T()

```
required init() {
    super.init()
}

override func setValue(value: AnyObject!, forUndefinedKey key: String) {
    switch key {
    case "content":
        content = value as! T
    default:
        print(" '\(value)' for key '\(key)' not handled.")
    }
}
```

}

**Comment by evermeer** [comment]
Hi,

Is "MaiTian.StatusModel" The string that you are passing on to the function? Could you try doing it without the MaiTian? So only pass 'StatusModel' to the swiftClassTypeFromString. EVReflection will extract the module name itself. So you don't need to pass it to the function. Currently If you want to use a class from a different bundle that the main bundle you have to call the setBundle function.

I do not have a check for passing a classname plus bundle name. Maybe I should have checked for a dot in the string. I will add that later this week.

... Edwin

**Comment by mengjiu-cao** [comment]
Hi,
thanks for your anwer.
when app infokey "CFBundleName" is Chinese string, it will crash.
 so I tried replacing code  var appName = bundle.infoDictionary?["CFBundleName"] as? String ?? "" to var appName = "MaiTian" , it will run without crash 
 "MaiTian.StatusModel" =>"MaiTian" is main bundle name, "StatusModel" is classname , it generated in function swiftClassTypeFromString  classStringName = "(appName).(className)"

but now when using generics  StatusModel<T: EVObject>, it crash.  the reason is also that swiftClassTypeFromString return nil

**Comment by evermeer** [comment]
I will have a look at it this evening (your night/morning)

**Comment by evermeer** [comment]
the line bundle.infoDictionary?["CFBundleName"] will try to get the bundle identifier from your mainBundle. So it looks like that in your case it's not able to get MaiTian from that.  What is that line returning? An empty string?

If you want to force this to work, then you could try calling:
EVReflection.setBundleIdentifier(StatusModel)
You only have to set that once. From then on It will always use the bundleIdentifier where the StatusModel is part of.

**Comment by evermeer** [comment]
Ah! I found a bug. The string representation for a generic class will start with _TtG instead of _TtC

I just added the test below to the WorkaroundSwiftGenericsTests and now it succeeds. I will release a new version in a moment.

This does mean that when you wan to use the swiftClassTypeFromString you can't use a string like  StatusModel or even MaiTian.StatusModel. If you wan to know what string to use, then see the swiftStringFromClass function

```
func testClassToAndFromString() {
    // Test the EVReflection class - to and from string
    let theObject = MyGenericObject<InstanceObject>()
    let theObjectString: String = EVReflection.swiftStringFromClass(theObject)
    NSLog("swiftStringFromClass = \(theObjectString)")

    let nsobject = EVReflection.swiftClassFromString(theObjectString)
    NSLog("object = \(nsobject)")
    XCTAssert(nsobject != nil, "Pass")
}
```

**Comment by evermeer** [comment]
I just released version 2.9.1
You do need to call the swiftClassTypeFromString with a string that starts with _TtG which is an internal string representation of a generic class. If you need to find out what string to use exactly, then call the function swiftStringFromClass like the unit test you see above.

If you have any other questions then just let me know.

**Comment by evermeer** [comment]
The reason why you need the _TtG notation instead of just "MaiTian.StatusModel" is that Swift needs to know the type of your generic class. I know it would be nicer if Swift would support something like "MaiTian.StatusModel<String>" but unfortunately it does not. Swift is using the _TtG notation for describing the exact type. In Swift 1.0 all object descriptions were in the _Tt format. Currently also nested objects are still in that format. Hopefully a future version of Swift will use a better format for all types.
